File: src/test/java/com/zalphion/featurecontrol/TestFixtures.java
package com.zalphion.featurecontrol;

import com.zalphion.featurecontrol.bundle.FeatureBundle;

import java.io.IOException;

public class TestFixtures {

    private TestFixtures() {}

    public static final FeatureBundle bundle1;

    static {
        try {
            bundle1 = FeatureBundle.fromClasspath("com/zalphion/featurecontrol/bundle1.json");
        } catch (IOException e) {
            throw new RuntimeException("Failed to load test bundle", e);
        }
    }
}

File: src/test/kotlin/com/zalphion/featurecontrol/TestFixtures.kt
package com.zalphion.featurecontrol

import com.zalphion.featurecontrol.source.FeatureSource
import kotlinx.serialization.ExperimentalSerializationApi
import kotlinx.serialization.json.Json
import kotlinx.serialization.json.decodeFromStream

@OptIn(ExperimentalSerializationApi::class)
internal val testBundle = FeatureSource::class.java.classLoader
    .getResourceAsStream("com/zalphion/featurecontrol/bundle1.json")!!
    .use { Json.decodeFromStream(FeatureBundle.serializer(), it) }